src/components/header/OverflowEdgeIndicator.js

```diff
@@ -2,21 +2,40 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// @flow
+
 import React, { PureComponent, PropTypes } from 'react';
 import classNames from 'classnames';
 
 import './OverflowEdgeIndicator.css';
 
+type Props = {
+  className: string,
+  children: number,
+};
+
 class OverflowEdgeIndicator extends PureComponent {
-  constructor(props) {
+  props: Props;
+  state: {
+    overflowsOnTop: ?boolean,
```

Comment:
You are right, I might misunderstand the error. I'll paste it here. If I remove the question marks in the state object I get the following error:

```
src/components/header/OverflowEdgeIndicator.js:76
 76:         className={classNames('overflowEdgeIndicator', className, this.state)}
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function call
 76:         className={classNames('overflowEdgeIndicator', className, this.state)}
                                                                       ^^^^^^^^^^ object type. This type is incompatible with
 14:     ...classes: Array<$npm$classnames$Classes>
                           ^^^^^^^^^^^^^^^^^^^^^^^ union: string | object type | array type | boolean literal `false` | undefined | null. See lib: flow-typed/npm/classnames_v2.x.x.js:14
  Member 1:
    5:   string |
         ^^^^^^ string. See lib: flow-typed/npm/classnames_v2.x.x.js:5
  Error:
   76:         className={classNames('overflowEdgeIndicator', className, this.state)}
                                                                         ^^^^^^^^^^ object type. This type is incompatible with
    5:   string |
         ^^^^^^ string. See lib: flow-typed/npm/classnames_v2.x.x.js:5
  Member 2:
    6:   {[className: string]: ?boolean } |
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ object type. See lib: flow-typed/npm/classnames_v2.x.x.js:6
  Error:
   22:     overflowsOnBottom: boolean,
                              ^^^^^^^ boolean. This type is incompatible with
    6:   {[className: string]: ?boolean } |
                               ^^^^^^^^ null. See lib: flow-typed/npm/classnames_v2.x.x.js:6
  Member 3:
    7:   Array<string> |
         ^^^^^^^^^^^^^ array type. See lib: flow-typed/npm/classnames_v2.x.x.js:7
  Error:
   76:         className={classNames('overflowEdgeIndicator', className, this.state)}
                                                                         ^^^^^^^^^^ object type. This type is incompatible with
    7:   Array<string> |
         ^^^^^^^^^^^^^ array type. See lib: flow-typed/npm/classnames_v2.x.x.js:7
  Member 4:
    8:   false |
         ^^^^^ boolean literal `false`. See lib: flow-typed/npm/classnames_v2.x.x.js:8
  Error:
   76:         className={classNames('overflowEdgeIndicator', className, this.state)}
                                                                         ^^^^^^^^^^ object type. This type is incompatible with
    8:   false |
         ^^^^^ boolean literal `false`. See lib: flow-typed/npm/classnames_v2.x.x.js:8
  Member 5:
    9:   void |
         ^^^^ undefined. See lib: flow-typed/npm/classnames_v2.x.x.js:9
  Error:
   76:         className={classNames('overflowEdgeIndicator', className, this.state)}
                                                                         ^^^^^^^^^^ object type. This type is incompatible with
    9:   void |
         ^^^^ undefined. See lib: flow-typed/npm/classnames_v2.x.x.js:9
  Member 6:
   10:   null
         ^^^^ null. See lib: flow-typed/npm/classnames_v2.x.x.js:10
  Error:
   76:         className={classNames('overflowEdgeIndicator', className, this.state)}
                                                                         ^^^^^^^^^^ object type. This type is incompatible with
   10:   null
         ^^^^ null. See lib: flow-typed/npm/classnames_v2.x.x.js:10


Found 1 error
```